Userland/Utilities/stat.cpp

```diff
@@ -72,16 +72,16 @@ static ErrorOr<int> stat(StringView file, bool should_follow_links)
 
     outln(")");
 
-    auto print_time = [](time_t t) {
-        outln("{}", Core::DateTime::from_timestamp(t).to_string());
+    auto print_time = [](timespec t) {
+        outln("{}.{}", Core::DateTime::from_timestamp(t.tv_sec).to_string(), t.tv_nsec);
```

Comment:
shouldn't this add padding?
1 nanosecond is not 0.1 second
Best would be printing a fixed point number
the nanoseconds use ~30 bit
time_t meanwhile is 64 bit (although c does not seem to dictate this?)
If we would have a number smaller that 32 bits we could use a `FixedPoint<30, i64>`

Reply:
Duh, good catch! Updated to use `{:09}` which covers the full `tv_nsec` range.